packages/runtime-common/schema-analysis-plugin.ts

```diff
@@ -36,13 +42,34 @@ export interface PossibleField {
 }
 
 export interface Options {
-  possibleCards: PossibleCardClass[];
+  possibleCards: PossibleCardClass[]; //cards may not be exports
+  exports: Export[];
```

Comment:
I've experienced one weird thing here and its that because I join these two arrays in module-syntax. There is no conformity with how the stuff is sorted in the file with the listing on the lhs "In this file"

Reply:
To fix this I have made this PR. https://github.com/cardstack/boxel/pull/708. This also includes a little bit of a refactor but should make things clearer in terms of types -- merging in.